### DISCUSSION TITLE
Auto Update Time

### DISCUSSION THREAD

**Discussion by insomniacj93** [Budibase/budibase#6721] [Help] [answered]
Hello, is there anyway i can update the time?
I'm currently using a javascript function to return the exact time but it only "updates" everytime i refresh the page, my intention is to have a real time without the need to refresh the page.

code:

function padTo2Digits(num) {
  return String(num).padStart(2, '0');  
}

const date = new Date();
const current_time =
  padTo2Digits(date.getHours()) + ':' + padTo2Digits(date.getMinutes());

return current_time;

Thank you in advance!

**Comment by R2bEEaton** [comment, marked as answer]
Unfortunately there's no supported method for doing this in Budibase. However, here's one example of how you could do this regardless:

Create an Embed element and paste in the following code to the Handlebar code:
```
<p id="updatingTime" class="">gaming</p>
<img style="display: none;" src onerror="
	function updateTime() {
		function padTo2Digits(num) {
		return String(num).padStart(2, '0');
		}

		const date = new Date();
		const current_time =
		padTo2Digits(date.getHours()) + ':' + padTo2Digits(date.getMinutes()) + ':' + padTo2Digits(date.getSeconds());

		document.getElementById('updatingTime').innerHTML = current_time;
	}
	updateTime();
	setInterval(function () {updateTime();}, 1000);
">
```
What does it do? Initializes a JavaScript setInterval to run every 1000 milliseconds and sets the innerHTML of the included `<p>` element. I added seconds so it's easier to visualize. The `<p>` element can be styled as desired or with Svelte classes to mimic Budibase elements.

Attached is an export of this example if you would like to inspect it further.
[Updating Time Example-export-1657913724917.txt](https://github.com/Budibase/budibase/files/9123551/Updating.Time.Example-export-1657913724917.txt)

This example is considered code injection, so be careful with this - it may look suspicious on a website.

**Reply by insomniacj93** [reply]
Thank you very much! @R2bEEaton 
It is working, I'm also giving the css a try. 
I believe we are restricted to the basic editing am i right?
No funny animations or anything like that?

**Reply by R2bEEaton** [reply]
No problem, @insomniacj93!
You can do whatever you want with custom CSS. If you want it to look more like a normal Budibase app, use the Svelte classes (use developer mode in most browsers to check the classes of the element you want to recreate the look of), but if you want to go make funny animations, go right ahead. Just add a `<style>` tag to the embed or `<link>` a stylesheet.
